services/website/website.service.js

```diff
@@ -38,118 +30,76 @@ const documentation = `
   <button onclick="(function(el) { el.value = decodeURIComponent(el.value); })(document.getElementById('websiteDocUrlField'))">Decode</button>
 </p>
 <p>
-  <code>[OPTIONS]</code> can be:
-  <ul>
-    <li>
-      Nothing:&nbsp;
-      <code>…/website/…</code>
-    </li>
-    <li>
-      Online and offline text:&nbsp;
-      <code>…/website-up-down/…</code>
-    </li>
-    <li>
-      Online and offline text, then online and offline colors:&nbsp;
-      <code>…/website-up-down-green-orange/…</code></li>
-    </li>
-  </ul>
-  <table class="centered"><tbody>
-    <tr><td>   Dashes <code>--</code>
-      </td><td>  →
-      </td><td>  <code>-</code> Dash
-    </td></tr>
-    <tr><td>   Underscores <code>__</code>
-      </td><td>  →
-      </td><td>  <code>_</code> Underscore <br/>
-    </td></tr>
-    <tr><td>   Slashes <code>//</code>
-      </td><td>  →
-      </td><td>  <code>/</code> Slash <br/>
-    </td></tr>
-    <tr><td>   <code>_</code> or Space <code>&nbsp;</code>
-      </td><td>  →
-      </td><td>  <code>&nbsp;</code> Space
-    </td></tr>
-  </tbody></table>
+  The messages and colors for the up and down states can also be customized.
 </p>
 `
 
-// This legacy service should be rewritten to use e.g. BaseJsonService.
-//
-// Tips for rewriting:
-// https://github.com/badges/shields/blob/master/doc/rewriting-services.md
-//
-// Do not base new services on this code.
-module.exports = class Website extends LegacyService {
+module.exports = class Website extends BaseService {
   static get category() {
     return 'monitoring'
   }
 
   static get route() {
     return {
-      base: '',
-      pattern: '',
+      base: 'website',
+      pattern: ':protocol(https|http)/:hostAndPath',
```

Comment:
We're able to use the standard pattern format here for `hostAndPath` because the badge requires end users to do the url encoding correct? 

Would it be possible to treat this like the other badge patterns that accept a `hostAndPath` and then do any required encoding?

Reply:
Hmm, the docs do say that, though I don't think it has to be encoded in order to work. The examples in the test do not encode the slashes, for example.

Reply:
One of the examples in the docs used paths, i.e. `http/www.website.com/path/to/page.html` which I think would have to be encoded or the route param would need to be `:hostAndPath+`

Is that a scenario we need to support?

Reply:
Ah, yes, it should be `hostAndPath+`. Sorry, I misunderstood when you first said that.